### ISSUE TITLE
Inconsistencies in validation accuracy

### ISSUE THREAD

**Issue by Wyatts1** [rdcolema/tensorflow-image-classification#9]
i removed the "if i % int(data.train.num_examples/batch_size) == 0: " statement for printing out the validation accuracy for each epoch so it is printed every iteration. I have a for loop calling optimize(iterations=1) immediately followed by print validation accuracy. The validation accuracy print in optimize and the one in print_validation_accuracy() give me different percentages. There is no other training going on between the two calls and yet they give different results. do you know why this might be

**Comment by rdcolema** [comment]
Hi, I haven't updated this script in a while but I think from taking a quick look that in the `optimize` function it's only printing accuracy for the most recent batch whereas `print_validation_accuracy` evaluates accuracy over the entire validation set. 

Would that explain the discrepancy you're seeing?

-RC

**Comment by Wyatts1** [comment]
I have been looking at it since posting and i came to the same conclusion. the discrepancy is that they are often completely different numbers. The more training i do the closer they get to 100% but sometimes it obviously takes awhile for the batches accuracies to 100%.
